### PR TITLE
UnusedPrivateField false positive #324

### PR DESCRIPTION
**Pull request by eeree**
Closes phpmd/phpmd#324

### CODE REVIEW

File: src/main/php/PHPMD/Rule/UnusedPrivateField.php [hunk 1]
```diff
@@ -176,15 +176,33 @@ protected function isValidPropertyNode(ASTNode $node = null)
...
         return (
             $owner->isInstanceOf('SelfReference') ||
             $owner->isInstanceOf('StaticReference') ||
             strcasecmp($owner->getImage(), '$this') === 0 ||
             strcasecmp($owner->getImage(), $class->getImage()) === 0
         );
     }
+
+    /**
+     * Looks for owner of given variable.
```
Comment (ravage84) [inline src/main/php/PHPMD/Rule/UnusedPrivateField.php]: "a given" or "the given" (I think it possible to omit this in English, just hate it 😁 )

File: src/main/php/PHPMD/Rule/UnusedPrivateField.php [hunk 2]
```diff
@@ -176,15 +176,33 @@ protected function isValidPropertyNode(ASTNode $node = null)
...
+     * @return \PHPMD\Node\ASTNode
+     */
+    protected function getOwner(ASTNode $postfix)
+    {
+        $owner = $postfix->getParent()->getChild(0);
+        if ($owner->isInstanceOf('PropertyPostfix')) {
+            $owner = $owner->getParent()->getParent()->getChild(0);
+        }
+
+        if ($owner->getParent()->isInstanceOf('ArrayIndexExpression')) {
```
Comment (ravage84) [inline src/main/php/PHPMD/Rule/UnusedPrivateField.php]: Why didn't you combine the two conditons as they lead to the same code?

Comment (eeree) [inline src/main/php/PHPMD/Rule/UnusedPrivateField.php]: They don't : ) Pay attention to one more `getParent()` call. It's most likely PDepend issue, but it's easily fixable here.

Comment (ravage84) [inline src/main/php/PHPMD/Rule/UnusedPrivateField.php]: I meant that the code WITHIN the if clauses was the same:
````
$owner = $owner->getParent()->getParent()->getChild(0);
````
Never mind it's readable and that's good.

**Pull request merged**